File: src/protocol.c
#include "pico/stdlib.h"

void format_packet(uint16_t value, uint8_t chan, uint8_t *buff_out) {
    buff_out[0] = (value >> 8) & 0xFE;
    buff_out[1] = (value >> 1) & 0xFE;
    buff_out[2] = ((value << 6) & 0xC0) | ((chan << 2) & 0x3C);
    buff_out[3] = 0x01;
}

File: micropython-port/protocol.py
def format_packet(value, channel):
    packet = bytes(4)
    packet[0] = (value >> 8) & 0xFE
    packet[1] = (value >> 1) & 0xFE
    packet[2] = ((value) << 6) & 0xC0) | ((channel << 2) & 0x3C)
    packet[3] = 0x01
    return packet
